src/compact.jl

```diff
@@ -68,20 +68,6 @@ for epoch in 1:1000
   Flux.train!((m,x,y) -> (m(x) - y)^2, model, data, optim)
 end
```

Comment:
```suggestion
end

When using `@compact` to hierarchically construct complex models, you may find it useful to specify a custom printout for the model using [`NoShow`](@ref).
```

Reply:
@mcabbott what do you think of adding this paragraph in the old one's stead?

Reply:
We could do... maybe my objection is that "`@compact` to hierarchically construct complex models" really seems like one example, no need for hierarchical `@compact` to want to tidy up. 

#20 wants bigger changes to the docstring. 

This suggestion also seems like it's before instead of after the code fence.

Reply:
I made a new suggestion below with a simpler sentence.